apps/backend/src/modules/users/user.service.ts

```diff
@@ -33,17 +45,36 @@ class UserService implements Service {
 
 		const { encryptedData, salt } = await encryption.encrypt(payload.password);
 
+		const group = await this.groupService.findByKey(GroupKey.USERS);
+
 		const item = await this.userRepository.create(
 			UserEntity.initializeNew({
 				email: payload.email,
 				firstName: payload.firstName,
+				groupId: group.id,
 				lastName: payload.lastName,
 				passwordHash: encryptedData,
 				passwordSalt: salt,
 			}),
 		);
 
-		return item.toObject();
+		if (!item.toObject().id) {
+			throw new UserError({
+				message: CommonExceptionMessage.COMMON_EXCEPTION_MESSAGE,
+				status: HTTPCode.INTERNAL_SERVER_ERROR,
+			});
+		}
+
+		const newUser = await this.userRepository.findById(item.toObject().id);
+
+		if (!newUser) {
+			throw new UserError({
+				message: CommonExceptionMessage.COMMON_EXCEPTION_MESSAGE,
+				status: HTTPCode.NOT_FOUND,
+			});
+		}
+
+		return newUser.toObject() as UserAuthResponseDto;
```

Comment:
As I understood, now we are returning a user with groups and permissions from the repository
```suggestion
		return item.toObject();
```